A note for empty string returned for default ports

ansballard opened the pull request:
#### Summary

Currently it's only noted that an empty string is returned when no port is included in the given url. This does not cover when a port is given, but with a protocol that has a matching default port. So `new URL('http://localhost:80').port` returns `''`, same with `new URL('https://localhost:443').port`, `new URL('ssh://localhost:22').port`, etc. Not sure if an example showing this should be added in the Examples section as well, but my assumption is that the note in the description is enough.

Also happy to take feedback on the wording, not sure if `given port number` is too explicit or `protocol's default` is too ambiguous.

#### Motivation

My assumption of what was returned when explicitly providing a port in a URL to parse was that it would populate regardless of protocol, so I think it's worthwhile to describe that difference.

#### Supporting details

_N/A_

#### Related issues

_N/A_

#### Metadata

- [ ] Adds a new document
- [X] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

ansballard committed:
A note for empty string returned for default ports

Currently it's only noted that an empty string is returned when no port is included in the given url. This does not cover when a port is given, but with a protocol that has a matching default port. So `new URL('http://localhost:80').port` returns `''`, same with `new URL('https://localhost:443').port`, `new URL('ssh://localhost:22').port`, etc. Not sure if an example showing this should be added in the Examples section as well, but my assumption is that the note in the description is enough.

Also happy to take feedback on the wording, not sure if `given port number` is too explicit or `protocol's default` is too ambiguous.

sideshowbarker commented:
Thanks for raising this — I think it might be require a longer explanation than what the patch here as-is currently provides.

What I mean is, https://developer.mozilla.org/en-US/docs/Web/API/URL/port is a property of the post-parsing URL object, and needs to be considered separate from the constructor.

So when we do `new URL('http://localhost:80')`, we end up with a URL object which looks like this:

```js
{
  href: "http://localhost/",
  origin: "http://localhost",
  protocol: "http:",
  username: "",
  password: "",
  host: "localhost",
  hostname: "localhost",
  port: "",
  pathname: "/",
  search: ""
}
```
That is, in that resulting URL object there’s no record of there having been a `:80` in the input string that was passed to the constructor.

So in that context — the context of the resulting URL object separate from the constructor, it doesn’t make sense to talk about _“the given port number”_, because there is no given port number at that point.

And so I think instead what’s needed here is to replace the entire _“If the URL does not contain an explicit port number, it will be set to `''`” sentence with something like this:

> If an input string passed to `URL()` constructor doesn’t contain an explicit port number (e.g., `https://localhost`) or contains a port number that’s the default port number corresponding to the protocol part of the input string (e.g., `https://localhost:443`), then in the URL object the constructor returns, the value of the `port` property will be `''` (the empty string).

ansballard committed:
updated the URL.port description

- updated the description of how the port is parsed based on PR review
- only change was to change how the empty string is references based on other pages

ansballard commented:
ty @sideshowbarker! I went ahead and copied your update in, aside from the last line about the empty string. It read a little off, so I found a couple other references to empty strings in other pages and used one of those. Have a great weekend!

sideshowbarker committed:
Copy edits

teoli2003 reviewed (approved):
👍
